pkg/vsphere/vm/vm.go

```diff
@@ -203,6 +203,32 @@ func (vm *VirtualMachine) getNetworkName(op trace.Operation, nic types.BaseVirtu
 	return nic.GetVirtualEthernetCard().DeviceInfo.GetDescription().Summary, nil
 }
 
+func (vm *VirtualMachine) GetMacAddressBasedSpecifiedNetworkName(ctx context.Context, ipaddress string) (string, error) {
+	op := trace.FromContext(ctx, "GetMacAddressBasedSpecifiedNetworkName")
+
+	var err error
+	var mvm mo.VirtualMachine
+
+	if err = vm.VirtualMachine.Properties(op, vm.Reference(), []string{"guest.net"}, &mvm); err != nil {
+		op.Errorf("Unable to get vm guest net info: %s", err)
+		return "", err
+	}
+
+	if mvm.Guest == nil || mvm.Guest.Net == nil {
+		return "", errors.New("Guest net info back nil, the vm was likely is not ready")
```

Comment:
Better to rephrase the error message to "Guest net info is nil. The vm is likely not ready"